src/Adapter/Product/Image/Uploader/ProductImageUploader.php

```diff
@@ -185,4 +197,37 @@ private function deleteCachedImages(Image $image): void
             }
         }
     }
+
+    /**
+     * Note: we can't delete the whole folder here, or Image::delete will return an error
+     * because it expects the original image and the folder to be present in order to remove
+     * them correctly. So for now this service only handles removing generated image types.
+     * When Image ObjectModel is no longer used, it could also remove the remaining files.
+     *
+     * @param string $imagePath
+     * @param array $imageTypes
+     *
+     * @throws CannotUnlinkImageException
+     */
+    private function deleteGeneratedImages(string $imagePath, array $imageTypes): void
+    {
+        $fileExtension = pathinfo($imagePath, PATHINFO_EXTENSION);
+        $destinationExtension = '.jpg';
+
+        foreach ($imageTypes as $imageType) {
+            $generatedImagePath = sprintf('%s-%s%s', rtrim($imagePath, '.' . $fileExtension), stripslashes($imageType->name), $destinationExtension);
```

Comment:
```suggestion
            $generatedImagePath = sprintf('%s-%s%s', rtrim($imagePath, '.' . $fileExtension), stripslashes($imageType->name), $destinationExtension);
```
Move `rtrim($imagePath, '.' . $fileExtension)` outside the foreach, it never change